src/dict.c

```diff
@@ -971,6 +973,7 @@ static long _dictKeyIndex(dict *d, const void *key, uint64_t hash, dictEntry **e
         return -1;
     for (table = 0; table <= 1; table++) {
         idx = hash & d->ht[table].sizemask;
+        if (table == 0 && idx < d->rehashidx) continue; 
```

Comment:
If non-rehashing, rehashidx is -1, it will overflow.
```suggestion
        if (table == 0 && (long)idx < d->rehashidx) continue; 
```

Reply:
Another problem, if dict is not in rehashing, continue will go to the second table, but at this time the second table is NULL.
In the old code it would be handled by `if (!dictIsRehashing(d)) return NULL;` at the bottom.

Reply:
> 
not rehashing don't (long)idx < d->rehashidx must be false, will not continue

Reply:
@ohye3166 Thx, You are right.

Reply:
> If non-rehashing, rehashidx is -1, it will overflow.
Thx for the suggestion.